services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/controller/SubnetController.java

```diff
@@ -523,8 +523,10 @@ public ResponseId deleteSubnetState(@PathVariable String projectId, @PathVariabl
             } else {
                 rangeId = ipV6RangeId;
             }
-//            Boolean checkIfAnyPortInSubnet = this.subnetService.checkIfAnyPortInSubnet(rangeId);
-//            if (checkIfAnyPortInSubnet) {
+
+            // TODO: check if there is any gateway / non-gateway port for the subnet, waiting for PM new API
+//            Boolean checkIfAnyNoneGatewayPortInSubnet = this.subnetService.checkIfAnyPortInSubnet(rangeId);
+//            if (checkIfAnyNoneGatewayPortInSubnet) {
 //                throw new HavePortInSubnet();
 //            }
```

Comment:
@kevin-zhonghao Please check PM's new API for port count in a subnet, it should be `{"/project/{project_id}/subnet-port-number/{subnet_id}`. This api will only return non-gateway port count. If it's not zero, the subnet cannot be deleted. If it returns zero, we need to ask PM to delete the gateway port first before delete the subnet.

Reply:
Sure